src/components/Table/docs/SortableTableDefault.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { FC, useCallback } from 'react';
+import { orderBy } from 'lodash';
+import { Table, TableCell, TableSortableHeaderCell, TableRow } from '..';
+import { Text } from '../..';
+import { useSortBy } from '../hooks/useSortBy';
+
+const data = [
+    {
+        id: 1,
+        name: 'Zaha',
+        emailAddress: 'zaha@example.com',
+        role: 'Admin',
+        costCenter: 'Design System'
+    },
+    {
+        id: 2,
+        name: 'Alex',
+        emailAddress: 'alex@example.com',
+        role: 'Booker',
+        costCenter: 'Product'
+    },
+    {
+        id: 3,
+        name: 'Britta',
+        emailAddress: 'britta@example.com',
+        role: 'Passenger',
+        costCenter: 'Customer'
+    },
+    {
+        id: 4,
+        name: 'Caio',
+        emailAddress: 'caio@example.com',
+        role: 'Admin',
+        costCenter: 'Design System'
+    }
+];
+
+export const SortableTableDefault: FC = () => {
+    const { sortBy, setSortBy } = useSortBy('id');
+    const handleSortingChange = useCallback(
+        field => {
+            setSortBy(field);
+        },
+        [setSortBy]
+    );
+
+    const direction = sortBy.direction?.toLowerCase() || 'asc';
+    // @ts-expect-error Argument of type 'string' is not assignable to parameter of type 'Many<boolean | "asc" | "desc"> | undefined'.ts(2769)
+    const sortedData = orderBy(data, sortBy.field, direction);
+
+    return (
+        <Table rowStyle="zebra" rowSize="small">
+            <thead>
+                <TableRow>
+                    <TableSortableHeaderCell
+                        field="id"
+                        active={sortBy.field === 'id'}
```

Comment:
Just a suggestion, I feel like we can make the usage a bit cleaner.
```tsx
                    <TableSortableHeaderCell
                        sortBy="name"
                        field={sortBy.field}
                        direction={sortBy.direction}
                        onSortChange={handleSortingChange}
                    >
```

if you write it like this, you can internally check for the active state by checking `sortBy === field` or something. The reason I propose is, in addition to it looking cleaner, is because I feel that the current way makes it a bit easier to use wrongly, for example with `active = true; direction = 'NONE'`. If we let the component decide this by itself, we can remove the ambiguity and also make the usage a little cleaner, unless you see any downsides

Reply:
I though about this... back then and now.

The reasoning is to have less logic embedded in the component so it can be used more flexibly. For instance, with the proposed API, it would be impossible to have a table sorted by 2 different columns at the same time. Whereas if you have full control from outside, you could achieve that fairly easily.

Just a reminder `direction` here can only be ASC or DESC – NONE is only used internally, for the sorting indicator.

Reply:
No strong opinions, feel free to go with what makes more sense to you, I think both are figure-outable enough
> it would be impossible to have a table sorted by 2 different columns at the same time
I don't think we have this anywhere, no? Sounds like a group-by kinda thing, but I have no idea how the UX will be if we want to show a doubly-sorted table, so I am not sure if it's very valid in this case but yeah the proposed API wouldn't work here unless we change the behaviour of the `sortBy.field`, _(I mean we probably have to do that anyway if we want to sort by two fields)_, imagine it being an array instead. We can do the active check by checking if the array contains the `sortBy` prop element.

I mean, I think we can do it both ways, I think if we can reduce "unnecessary" moving parts then it will in turn reduce the complexity of the component's API, and make it less likely to have buggy behaviour. But I leave it upto you, I think both are readable approaches

Reply:
Having a flexible usage, without much embedded logic will definitely support more cases, and I don't see this could lead to "buggy" usage, right now. 

This part of the logic clearly can vary between implementations (it's not required to use the useSortBy hook), so it shouldn't be the component's responsibility.